packages/node/src/handlers.ts

```diff
@@ -358,36 +359,49 @@ export function trpcMiddleware(options: SentryTrpcMiddlewareOptions = {}) {
       sentryTransaction.setContext('trpc', trpcContext);
     }
 
-    function captureError(e: unknown): void {
-      captureException(e, scope => {
-        scope.addEventProcessor(event => {
-          addExceptionMechanism(event, {
-            handled: false,
+    function shouldCaptureError(e: unknown): boolean {
+      if (typeof e === 'object' && e && 'code' in e) {
+        // Is likely TRPCError - we only want to capture internal server errors
+        return e.code === 'INTERNAL_SERVER_ERROR';
+      } else {
+        // Is likely random error that bubbles up
+        return true;
+      }
+    }
+
+    function handleErrorCase(e: unknown): void {
+      if (shouldCaptureError(e)) {
+        captureException(e, scope => {
+          scope.addEventProcessor(event => {
+            addExceptionMechanism(event, {
+              handled: false,
+            });
+            return event;
           });
-          return event;
-        });
 
-        return scope;
-      });
+          return scope;
+        });
+      }
     }
 
-    try {
-      return await next();
-    } catch (e: unknown) {
-      if (typeof e === 'object' && e) {
-        if ('code' in e) {
-          // Is likely TRPCError - we only want to capture internal server errors
-          if (e.code === 'INTERNAL_SERVER_ERROR') {
-            captureError(e);
-          }
-        } else {
-          // Is likely random error that bubbles up
-          captureError(e);
-        }
-      }
+    let maybePromiseResult;
```

Comment:
not directly related and nothing to do, but I feel like we've written this or very similar code a lot of times by now, maybe we can find a way to extract this into some util function that we can reuse 🤔